File: 백준/Silver/2751. 수 정렬하기 2/수 정렬하기 2.rs
use std::{
    io::{stdin, Read},
    str::FromStr,
};

fn main() {
    truncate_line();
    let mut nums = input_all_splitted_as::<i32>();
    nums.sort();

    let output = nums
        .iter()
        .map(|s| s.to_string())
        .collect::<Vec<String>>()
        .join("\n");

    println!("{output}");
}

fn truncate_line() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();
}

fn input_all_splitted_as<T>() -> Vec<T>
where
    T: FromStr,
    <T as FromStr>::Err: std::fmt::Debug,
{
    let mut buf = String::new();
    stdin().read_to_string(&mut buf).unwrap();

    buf.trim()
        .split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<_>>()
}

File: 백준/Silver/2751. 수 정렬하기 2/수 정렬하기 2.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n;
  cin >> n;
  vector<int> nums(n);

  for (int i = 0; i < n; i++) {
    cin >> nums[i];
  }

  sort(nums.begin(), nums.end());

  for (int num : nums) {
    cout << num << '\n';
  }

  return 0;
}